### PR TITLE
fix(vulnerabilities)/vulnerabilities_fixes_datahub

### PR DESCRIPTION
**Pull request by Sejal-NucleusTeq**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -89,7 +89,7 @@ project.ext.externalDependency = [
     'elasticSearchRest': 'org.elasticsearch.client:elasticsearch-rest-high-level-client:' + elasticsearchVersion,
     'elasticSearchTransport': 'org.elasticsearch.client:transport:' + elasticsearchVersion,
     'findbugsAnnotations': 'com.google.code.findbugs:annotations:3.0.1',
-    'graphqlJava': 'com.graphql-java:graphql-java:' + graphQLJavaVersion,
+    'graphqlJava': 'com.graphql-java:graphql-java:19.4',
```
Comment (david-leifker) [inline build.gradle:92]: The version here should be incremented above from `ext.graphQLJavaVersion = '19.0'` to the 19.4 version.

File: build.gradle [hunk 2]
```diff
@@ -202,7 +202,13 @@ project.ext.externalDependency = [
     'testContainersCassandra': 'org.testcontainers:cassandra:' + testContainersVersion,
     'typesafeConfig':'com.typesafe:config:1.4.1',
     'wiremock':'com.github.tomakehurst:wiremock:2.10.0',
-    'zookeeper': 'org.apache.zookeeper:zookeeper:3.4.14'
+    'zookeeper': 'org.apache.zookeeper:zookeeper:3.4.14',
+    'wire': 'com.squareup.wire:wire-compiler:3.7.1',
+    'charle':  'com.charleskorn.kaml:kaml:0.53.0',
+    'jsonv': 'org.json:json:20230227',
```
Comment (david-leifker) [inline build.gradle:208]: Looks like this is duplicated above `'json': 'org.json:json:20230227',` is the same as this one. Please consolidate to one.

File: datahub-upgrade/build.gradle [hunk 3]
```diff
@@ -23,6 +25,9 @@ dependencies {
     exclude group: "org.apache.htrace", module: "htrace-core4"
     exclude group: "org.eclipse.jetty", module: "jetty-util"
     exclude group: "org.apache.hadoop.thirdparty", module: "hadoop-shaded-protobuf_3_7"
+    exclude group: "com.charleskorn.kaml",module:"kaml"
```
Comment (david-leifker) [inline datahub-upgrade/build.gradle:28]: Couple of linting issues here, please add a space after the `,` to match the others. Thanks!

File: metadata-service/factories/build.gradle [hunk 4]
```diff
@@ -42,7 +42,13 @@ dependencies {
   testCompile externalDependency.mockito
   testCompile externalDependency.testng
   testCompile externalDependency.hazelcastTest
+  implementation externalDependency.jline
+  implementation externalDependency.common
+}
 
+configurations.all{
+  exclude group: "commons-io",module:"commons-io"
```
Comment (david-leifker) [inline metadata-service/factories/build.gradle:50]: Similar to above linting command, please add a space after the comma.

File: metadata-service/graphql-servlet-impl/build.gradle [hunk 5]
```diff
@@ -14,5 +14,14 @@ dependencies {
...
 
-  annotationProcessor externalDependency.lombok
+  implementation externalDependency.charle
+  implementation externalDependency.jsonv
+  implementation externalDependency.jetbrains
+
 }
+
+configurations.all{
+  exclude group: "com.charleskorn.kaml",module:"kaml"
```
Comment (david-leifker) [inline metadata-service/graphql-servlet-impl/build.gradle:24]: linting

File: metadata-service/war/build.gradle [hunk 6]
```diff
@@ -36,8 +36,14 @@ dependencies {
   runtime externalDependency.log4j2Api
   runtime externalDependency.logbackClassic
   implementation externalDependency.awsMskIamAuth
-}
 
+  implementation externalDependency.jsonv
+  implementation externalDependency.charle
+}
+configurations.all{
+  exclude group: "com.charleskorn.kaml",module:"kaml"
```
Comment (david-leifker) [inline metadata-service/war/build.gradle:44]: linting

**Comment by david-leifker** [comment]
Overall this looks good, just a few minor updates. If you enable us to modify the branch I can fix up the minor issues, otherwise look forward to a slight revision. Thank you for this!

**Comment by Sejal-NucleusTeq** [comment]
Hey David-leifker

Changes are made according to the reviews.

Thankyou
Sejal Jain
Nucleus Teq

**Comment by david-leifker** [comment]
Some of these fixes may have already been introduced to master, I suspect this is the branch conflicts. If possible please review the conflicts so that if there are remaining changes, we can merge them. Thank you!

**Comment by Sejal-NucleusTeq** [comment]
okey sure

**Comment by Sejal-NucleusTeq** [comment]
Hello David-leifker

**Review by david-leifker — APPROVED** [review]

**Comment by david-leifker** [comment]
Merged to alternative branch so I can make sure all the tests are passing for before merging. Plan is to merge from that branch. https://github.com/datahub-project/datahub/pull/8189

**Comment by Sejal-NucleusTeq** [comment]
Okay

**Pull request merged**